--- postgresql-16.spec
+++ postgresql-16.spec
@@ -86,6 +86,10 @@ Patch1:		%{sname}-%{pgmajorversion}-rpm-pgsql.patch
 Patch3:		%{sname}-%{pgmajorversion}-conf.patch
 Patch5:		%{sname}-%{pgmajorversion}-var-run-socket.patch
 Patch6:		%{sname}-%{pgmajorversion}-perl-rpath.patch
+Patch15:	pg%{pgmajorversion}-015-attoptions.diff
+Patch20:	pg%{pgmajorversion}-020-LOG-to-DEBUG1.diff
+Patch25:	pg%{pgmajorversion}-025-logical_commit_clock.diff
+Patch30:	pg%{pgmajorversion}-030-per-subtrans-commit-ts.diff
 
 BuildRequires:	perl glibc-devel bison flex >= 2.5.31
 BuildRequires:	gcc-c++
@@ -454,6 +458,10 @@ benchmarks.
 %patch -P 3 -p0
 %patch -P 5 -p0
 %patch -P 6 -p0
+%patch -P 15 -p1
+%patch -P 20 -p1
+%patch -P 25 -p1
+%patch -P 30 -p1
 
 %{__cp} -p %{SOURCE12} .
 
